src/http.rs

```diff
@@ -0,0 +1,22 @@
+use curl;
+use oauth2::*;
+use app::App;
+
+pub fn github(app: &App, url: &str, auth: &Token)
```

Comment:
This function might be a lot nicer to call if it was something along the lines of:

```rust
pub fn github<T: Decodable>(app: &App, url: &str, auth: &Token)
        -> Result<T, curl::ErrCode> { 
    // ...
}
```

Basically something that does the decoding for you automatically and fills in an error for bad JSON and bad utf-8-ness